core/ui/compose/designsystem/src/test/kotlin/app/k9mail/core/ui/compose/designsystem/atom/textfield/PasswordTextFieldOutlinedKtTest.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTextInput
+import app.k9mail.core.ui.compose.designsystem.R
+import app.k9mail.core.ui.compose.testing.ComposeTest
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import org.junit.Test
+
+class PasswordTextFieldOutlinedKtTest : ComposeTest() {
+
+    @Test
+    fun `should toggle password visibility`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+            )
+        }
+
+        val showPasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_show_password),
+        )
+        val hidePasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_hide_password),
+        )
+
+        // Check initial state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+
+        // Toggle password visibility
+        showPasswordNode.performClick()
+
+        // Check toggled state
+        showPasswordNode.assertDoesNotExist()
+        hidePasswordNode.assertIsDisplayed()
+
+        // Toggle password visibility
+        hidePasswordNode.performClick()
+
+        // Check toggled state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+    }
+
+    @Test
+    fun `should display password when toggled`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+            )
+        }
+
+        onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_show_password),
+        ).performClick()
+
+        onNodeWithText(VALUE).assertIsDisplayed()
+    }
+
+    @Test
+    fun `should not toggle password visibility when disabled`() = runComposeTest {
+        setContent {
+            PasswordTextFieldOutlined(
+                value = VALUE,
+                onValueChange = ON_VALUE_CHANGE,
+                enabled = false,
+            )
+        }
+
+        val showPasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_show_password),
+        )
+        val hidePasswordNode = onNodeWithContentDescription(
+            getString(R.string.core_ui_designsystem_accessibility_hide_password),
+        )
+
+        // Check initial state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+
+        // Toggle password visibility
+        showPasswordNode.performClick()
+
+        // Check toggled state
+        showPasswordNode.assertIsDisplayed()
+        hidePasswordNode.assertDoesNotExist()
+        onNodeWithText(VALUE).assertDoesNotExist()
```

Comment:
This check should probably be performed every time the icon visibility is checked. Whether or not the password is visible is probably more important than the icon state.

It took me a while to understand what this does. So the assertion itself should probably be moved to a function with a descriptive name.

Reply:
I rewrote the test, after removing shared TextField tests